File: legacy/src/main/rs/emulate/legacy/archive/CompressionType.java
package rs.emulate.legacy.archive;

/**
 * The two types of archive compression support by the client.
 *
 * @author Major
 */
public enum CompressionType {

	/**
	 * The compression type where entire archive is compressed with bzip2, with only the 6 header bytes uncompressed.
	 */
	ARCHIVE_COMPRESSION,

	/**
	 * The compression type where meta information (e.g. archive entry identifiers) is not compressed, and only the
	 * actual entry data is.
	 */
	ENTRY_COMPRESSION,

	/**
	 * The compression type where neither the archive nor the entries are compressed. This is not used by the client,
	 * and should only be used for testing purposes.
	 */
	NO_COMPRESSION;

}

File: modern/src/main/kotlin/rs/emulate/modern/codec/CompressionType.kt
package rs.emulate.modern.codec

enum class CompressionType(val id: Int) {
    NONE(0),
    BZIP2(1),
    GZIP(2),
    LZMA(3);

    companion object {
        private val TYPES = values().associateBy(CompressionType::id)
        operator fun get(id: Int): CompressionType? = TYPES[id]
    }
}
